### PR TITLE
AIP 4844 + ArbOS 20

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        );
+
+        // set the new challenge manager impl
+        TransparentUpgradeableProxy challengeManager = TransparentUpgradeableProxy(
+            payable(address(l1AddressRegistry.rollup().challengeManager()))
+        );
+        govProxyAdmin.upgradeAndCall(
+            challengeManager,
+            newChallengeManagerImpl,
+            abi.encodeWithSelector(IChallengeManagerUpgradeInit.postUpgradeInit.selector, newOsp)
```
Comment (gzeoneth) [inline src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:106]: usually prefer to use encodeCall for type check

Comment (yahgwai) [reply src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol]: updated

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol
Comment (gzeoneth) [inline src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:1]: we also need to update the template in ValidatorWalletCreator

**Review by gzeoneth — CHANGES_REQUESTED** [review]

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol
Comment (gzeoneth) [reply src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:1]: nvm, ValidatorWallet cannot use blob anyway (we allow tx.origin only)

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+interface ISeqInboxPostUpgradeInit {
+    function postUpgradeInit() external;
+}
+
+interface IChallengeManagerUpgradeInit {
+    function postUpgradeInit(address _newOsp) external;
+    function osp() external returns (address);
+}
+
+/// @notice Upgrades the an arbitrum chain in preparation for 4844
```
Comment (gvladika) [inline src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:17]: Extra 'the'

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        TransparentUpgradeableProxy challengeManager = TransparentUpgradeableProxy(
+            payable(address(l1AddressRegistry.rollup().challengeManager()))
+        );
+        govProxyAdmin.upgradeAndCall(
+            challengeManager,
+            newChallengeManagerImpl,
+            abi.encodeCall(IChallengeManagerUpgradeInit.postUpgradeInit, (newOsp))
+        );
+
+        require(
```
Comment (gvladika) [inline src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:107]: Would it be worth to also check `govProxyAdmin.getProxyImplementation(challengeManager) == newChallengeManagerImpl` ?

**Review by gvladika — APPROVED** [review]
LGTM

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol (hunk 3 above)
Comment (yahgwai) [reply src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:107]: added check

File: src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol (hunk 2 above)
Comment (yahgwai) [reply src/gov-action-contracts/AIPs/AIP4844/AIP4844Action.sol:17]: updated

**Review by gzeoneth — APPROVED** [review]
LGTM

**Pull request merged**